### PR TITLE
Add new docs and autogeneration process for github pages based docs hosting

### PR DESCRIPTION
**Pull request by sungo**
To be clear, this is not intended to be complete documentation, but rather a base that we can start adding to.

### CODE REVIEW

File: docs/README [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Yeah, there's ruby stuff in here. You can ignore every bit of it unless you'd
```
Comment (karenetheridge) [inline docs/README]: ruby???

Comment (sungo) [reply docs/README]: please see the rest of the file :) jekyll is ruby. if you want to use jekyll to review the changes you've made, you can. or you don't have to.

**Comment by sungo** [comment]
@karenetheridge No, it shouldn't. Jekyll / GHpages renders html versions of the markup. That link points to the html-ized version. if you link to the `.md` version, the user will get the markdown text file which is not what I want there.

**Comment by karenetheridge** [comment]
Looks like this didn't incorporate the changes I made in my mail PR.

**Pull request merged**